### ISSUE TITLE
Highly fluctuating CPU %

### ISSUE THREAD

**Issue by biona001** [tro3/ThreadPools.jl#19]
On 8 cores with 8 threads, CPU % fluctuates a lot (100%-800%) when using `ThreadPools.@qthreads`, while `Threads.@threads` exhibit very consistent CPU%. There are no hyperthreading and BLAS threads are set to 1. 

Using `ThreadPools.@qthreads` (CPU % fluctuates a lot):
<img width="190" alt="threadpools" src="https://user-images.githubusercontent.com/16760873/89245075-e6bdfc80-d5bc-11ea-99f0-2ecc34df3f2d.png">

Using `Threads.@threads` (CPU % is consistent, and drops when 1 thread finishes all assigned tasks):
<img width="190" alt="threads" src="https://user-images.githubusercontent.com/16760873/89245081-e9205680-d5bc-11ea-9e85-c01ba9297613.png">

Sorry, no MWE as I'm testing it on my package which runs complicated routine. I wonder if this is expected behavior or some performance penalty for using `@qthreads`?

**Comment by biona001** [comment]
Please feel free to close if the description is too vague.

**Comment by tro3** [comment]
Kind of makes sense - `@threads` and `@bthreads` use a static scheduler.  They see how many jobs you have, divide them up evenly by job count, and start the threads.  `@qthreads` and `@qbthreads` have a queued scheduler.  All the jobs are put into a pool, and as one job finishes on one thread, it is replaced with the next one.  If some jobs are very long and some are very short, this can be more efficient.  If all the jobs are the same, or all the jobs are different but very short, it is less efficient.

What is the nature of the jobs you are running?  Note if you use `@logthreads` and `@logqthreads`, you'll be able to see how your jobs are being assigned and executed.

```julia
julia> pool = @logthreads for job in jobs
         myfunc(job)
       end;

julia> plot(pool)
```

**Comment by biona001** [comment]
Some of my jobs are really fast but others are really long, that's why I decided to try out `@qthreads`. Overall using `@qthreads` is faster than `@threads`, but when using the static scheduler some threads finish rather quickly while other threads may run for 2-3x longer. 

Still, the CPU % makes me uncomfortable.... Unfortunately, I tried to make a MWE mimicking my real application, to no avail (CPU % remains above 700% for both cases). I'm closing this issue for now since there's really no way to pinpoint what the cause is.
